lib/appsignal/tesla.ex

```diff
@@ -0,0 +1,188 @@
+  # when `Tesla.Middleware.PathParams` is present.
+  #
+  # If the URL does not contain a host, and a base URL is provided by
+  # `Tesla.Middleware.BaseUrl`, then the base URL's scheme, host,
+  # port and path are used instead.
+  # When using the base URL, its path fragment will be used, regardless
+  # of the value of `use_full_path`.
+  # If `use_full_path` is true, then the path fragments of the base URL
+  # and the URL are concatenated together.
+  def sanitise_url(url, base_url, use_full_path) do
+    %URI{
+      scheme: scheme,
+      host: host,
+      port: port,
+      path: path
+    } = URI.parse(url || "")
+
+    # Users can specify a host in the request's URL, which overrides the
+    # base URL in the BaseURL middleware. Only use the base URL if the URL
+    # in the telemetry event does not already have a host.
+    # If the BaseURL middleware is used before the Telemetry middleware,
+    # the telemetry event's URL will already have the base URL interpolated
+    # into it, and so the base URL argument will never be used.
+    if host == nil do
+      %URI{
+        scheme: base_scheme,
+        host: base_host,
+        port: base_port,
+        path: base_path
+      } = URI.parse(base_url || "")
+
+      full_path =
+        if use_full_path do
+          "#{base_path}#{path}"
+        else
+          base_path
+        end
+
+      URI.to_string(%URI{scheme: base_scheme, host: base_host, port: base_port, path: full_path})
+    else
+      full_path = if use_full_path, do: path
+
+      URI.to_string(%URI{scheme: scheme, host: host, port: port, path: full_path})
+    end
+  end
+
+  def env_middlewares(env) do
+    # Use the same middleware order implemented in `Tesla.prepare`, but ignore
+    # `post` client middlewares, which are not relevant to our use case:
+    # https://github.com/elixir-tesla/tesla/blob/9fbf221/lib/tesla.ex#L116-L120
+    normalize_middlewares(client_pre_middlewares(env) ++ module_middlewares(env))
+  end
+
+  defp module_middlewares(env) do
+    module = Map.get(env, :__module__)
+
+    if module != nil and function_exported?(module, :__middleware__, 0) do
+      module.__middleware__
+    end || []
+  end
+
+  defp client_pre_middlewares(env) do
+    client = Map.get(env, :__client__)
+
+    if client != nil do
+      Map.get(client, :pre, [])
+    end || []
+  end
+
+  defp normalize_middlewares(middlewares) do
+    # The `Tesla.Env.runtime()` type obtained from the `__middleware__`
+    # property in the module and the `pre` property in the client can
+    # contain middlewares and adapters of different shapes. Keep only
+    # module-based middlewares and their arguments, if any, using an
+    # unified value shape.
+
+    middlewares
+    |> Enum.map(fn
+      {:fn, _} -> nil
+      {middleware, _, opts} -> {middleware, opts}
+      {middleware, _} -> {middleware, nil}
+      _ -> nil
+    end)
+    |> Enum.reject(&(&1 == nil))
```

Comment:
There's a function to check for nils that you could use here: 

``` elixir
Enum.reject(&is_nil/1)
```

Reply:
Thank you!